roles/variables/defaults/main.yml

```diff
@@ -215,6 +215,21 @@ certificate_authority_expiration_days: 365
 ### Boolean to enable mTLS Authentication on all components. Configures all components to use mTLS for authentication into Kafka
 ssl_mutual_auth_enabled: false
 
+### mTLS server's config to enforce ssl client authentication. Options are none, requested, required
+ssl_client_authentication: >-
+  {%- if ssl_mutual_auth_enabled|bool -%}
+    {%- if deployment_strategy == 'parallel' -%}
+      required
+    {%- elif deployment_strategy in ['serial', 'rolling'] -%}
+      requested
+    {%- endif -%}
+  {%- else -%}
+      none
+  {%- endif -%}
+
+# yamllint disable-line rule:key-duplicates
+ssl_mutual_auth_enabled: "{{ true if ssl_client_authentication in ['required', 'requested'] else false }}"
```

Comment:
in config validations check if user is defining both these in contrdictory way and stop the setup